backend/src/main.py

```diff
@@ -35,102 +28,159 @@
 @asynccontextmanager
 async def lifespan(app: FastAPI):  # noqa: ARG001
     logger.info("startup: triggered")
+    await sync_to_async(
+        seed_database
+    )()  # for dev only, remove this line to disable seeding when deploying
```

Comment:
This should be way more robust. Why not just use find_spec or the settings.debug level to drive this?